### DISCUSSION TITLE
`<CR>` inserts `<C-R>=UltiSnips#ExpandSnippet()<CR>` with nvim-cmp and Ultisnips

### DISCUSSION THREAD

**Discussion by fidelski** [nvim-orgmode/orgmode#843] [Q&A] [answered]
I have a weird problem I cannot break down that is somewhat related to nvim-cmp and Ultisnips in conjunction with nvim-orgmode and my settings. I only encounter this using nvim-orgmode and it never appears in any other content. Whenever I open an org file and enter `<CR>` somewhere in the file I get the literal text `<C-R>=UltiSnips#ExpandSnippet()<CR>` inserted into the document. I could find a related issue in #799 and when I print the content of `b.org_old_cr_mapping` I get:

```
b:org_old_cr_mapping  {'lhs': '<CR>', 'mode': 'i', 'expr': 0, 'sid': 8, 'lnum': 67, 'noremap': 1, 'nowait': 0, 'rhs': '<C-R>=UltiSnips#ExpandSnippet()<CR>', 'lhsraw': '^M',
 'abbr': 0, 'script': 0, 'mode_bits': 16, 'silent': 1, 'buffer': 0, 'scriptversion': 1}
```

I cannot see `callback` set and `rhs` is not empty. Is there anything I could do to figure out what's going wrong here?

**Comment by kristijanhusak** [comment]
Can you tell me what do you get for `:imap <CR>` in a non orgmode buffer?

**Reply by fidelski** [reply]
That gives me:
```
i  <CR>        * <Lua 69: ~/.local/share/nvim/lazy/nvim-cmp/lua/cmp/utils/keymap.lua:133>
                 cmp.utils.keymap.set_map
```

**Comment by fidelski** [comment, marked as answer]
Oh boy! I accidently configured `<CR>` twice :see_no_evil:  - once in the configuration for Ultisnips:
```lua
 vim.g.UltiSnipsExpandTrigger = '<CR>'
```
**and** in:
```lua
cmp.setup ({
...
mapping = ...
                ['<CR>'] = cmp.mapping.confirm ({
                    behavior = cmp.ConfirmBehavior.Replace,
                    select = false
                }),
```

Removing the configuration of `vim.g.UltiSnipsExpandTrigger = '<CR>'` solves the problem!
